pkg/api/cluster_client.go

```diff
@@ -80,7 +81,11 @@ func (c ClusterClient) ReadByName(ctx context.Context, projectId, name string) (
 	}
 
 	if len(clusters.Data) != 1 {
-		return &models.Cluster{}, Error404
+		if most_recent {
+			sort.Slice(clusters.Data, func(i, j int) bool { return clusters.Data[i].CreatedAt.Seconds > clusters.Data[j].CreatedAt.Seconds })
+		} else {
+			return &models.Cluster{}, ErrorClustersSameName
```

Comment:
should `most_recent` be a default option if multiple clusters found? 
may be print a log saying that
`we found multiple clusters but returned the most recent one?`

Reply:
the reason im thinking is we made the api-call, we got the data and we are dropping it at the last min and asking the user to make another call?

Reply:
That's a good suggestion @dhilipkumars. 
In our current design, the client sends the return value and the errors. When we find two or more clusters with the same name, we can return this info from the client and turn it into a `Warning` level Diagnostic, but I need to think about how to do that. 

For now, the best I can offer is creating a ticket for this and adding a `tflog.Info` when most_recent is true.

Reply:
Actually, I'm not sure about the `tflog.Info` there because it's the client code. We're trying to keep the client code as terraform-agnostic as possible. So, I will write the debug info in the ticket as well.

Reply:
I created #121 so that we can tackle the problem in a dedicated time. 
Do you agree with that @dhilipkumars? If It's not a blocker, I'd like to merge this PR first, and then tackle the Warning system later.

Reply:
Thanks @serdardalgic im resolving this comment